Run and fix 4.2.0 API integration tests

|Wazuh version|Component|
|---|---|
| [4.2.0](https://github.com/wazuh/wazuh/tree/4.2) | API |

Hi team,

We must run all API integration tests to check all of them are passing in the latest 4.2.0 branch. 

After this run, update all the API integration tests failing.

With the last nightly Jenkins run (https://ci.wazuh.info/job/Test_integration_endpoints/697/), we have some tests failing in `master` (there are no recent `4.2.0` nightlies). Check them as they may also fail in 4.2.0.

Regards,
Manuel.

## Issue update

I have passed all API integration tests in local. All tests have also been passed here https://ci.wazuh.info/job/Test_integration_endpoints/720/.

In both iterations, the only API integration test failing is `test_task_endpoint`:

- Test failing

```
test_task_endpoints.tavern.yaml 
	 1 failed, 1 passed, 4 warning
```
	 

Error:

```
E   tavern.util.exceptions.TestFailError: Test 'Try to get tasks using select parameter' failed:
    - Error calling validate function '<function test_select_key_affected_items at 0x7f1987e17700>':
        Traceback (most recent call last):
          File "/usr/local/lib/python3.8/dist-packages/tavern/response/base.py", line 141, in _maybe_run_validate_functions
            vf(response)
          File "/usr/local/lib/python3.8/dist-packages/tavern/schemas/extensions.py", line 123, in inner
            return func(response, *args, **kwargs)
          File "/tmp/Test_integration_endpoints_B720_test_files/api/test/integration/tavern_utils.py", line 59, in test_select_key_affected_items
            assert (set1 == set() or set1 == set1.intersection({'id', 'agent_id', 'file'} | main_keys)), \
        AssertionError: Select keys are {'task_id'}, but the response contains these keys: {'create_time', 'status', 'command', 'agent_id'}
```

We need to add 'task_id', 'create_time', 'status', 'command' and 'agent_id' to the assert of the `test_select_key_affected_items` function:

`assert (set1 == set() or set1 == set1.intersection({'id', 'agent_id', 'file', 'task_id', 'create_time', 'status', 'command', 'agent_id'} | main_keys)), f'Select keys are {main_keys}, but the response contains these keys: {set1}'`

This is needed as the `min_select_fields` of task are:

`min_select_fields = {'task_id', 'agent_id', 'status', 'command', 'create_time'}
`
This will change in master and only `task_id` will be the `min_select_field` of task.

- Whole list of tests passing:

```
test_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_black_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_agent_DELETE_endpoints.tavern.yaml 
	 6 passed, 8 warnings

test_agent_GET_endpoints.tavern.yaml 
	 91 passed, 93 warnings

test_agent_POST_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_agent_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
	 
test_rbac_white_agent_endpoints.tavern.yaml 
	 40 passed, 42 warnings

test_rbac_black_agent_endpoints.tavern.yaml 
	 40 passed, 42 warnings

test_syscollector_endpoints.tavern.yaml 
	 159 passed, 161 warnings
	 
test_rbac_black_syscollector_endpoints.tavern.yaml 
	 9 passed, 11 warnings

test_rbac_white_syscollector_endpoints.tavern.yaml 
	 9 passed, 11 warnings
	 
test_decoder_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_rbac_black_decoder_endpoints.tavern.yaml 
	 6 passed, 8 warnings

test_rbac_white_decoder_endpoints.tavern.yaml 
	 6 passed, 8 warnings

test_rbac_black_rule_endpoints.tavern.yaml 
	 8 passed, 10 warnings

test_rbac_white_rule_endpoints.tavern.yaml 
	 8 passed, 10 warnings

test_rule_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_rbac_black_vulnerability_endpoints.tavern.yaml
	 1 passed, 3 warnings
	
test_vulnerability_endpoints.tavern.yaml
	 1 passed, 3 warnings
	 
test_rbac_white_vulnerability_endpoints.tavern.yaml
	 1 passed, 3 warnings
	 
test_rbac_black_syscheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings

test_rbac_white_syscheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings

test_syscheck_endpoints.tavern.yaml 
	 35 passed, 37 warnings

test_rbac_black_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_rbac_white_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_security_DELETE_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_security_GET_endpoints.tavern.yaml 
	 18 passed, 20 warnings

test_security_POST_endpoints.tavern.yaml 
	 7 passed, 9 warnings

test_security_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings

test_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_ciscat_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_cluster_endpoints.tavern.yaml 
	 44 passed, 1 xpassed, 47 warnings

test_experimental_endpoints.tavern.yaml 
	 10 passed, 1 xpassed, 13 warnings

test_logtest_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_manager_endpoints.tavern.yaml 
	 31 passed, 33 warnings

test_mitre_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_overview_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_black_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_rbac_black_ciscat_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_black_cluster_endpoints.tavern.yaml 
	 17 passed, 19 warnings

test_rbac_black_experimental_endpoints.tavern.yaml 
	 11 passed, 13 warnings

test_rbac_black_logtest_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_black_manager_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_rbac_black_mitre_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_black_overview_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_black_rootcheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings

test_rbac_black_sca_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_black_task_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_rbac_white_ciscat_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_cluster_endpoints.tavern.yaml 
	 17 passed, 19 warnings

test_rbac_white_experimental_endpoints.tavern.yaml 
	 11 passed, 13 warnings

test_rbac_white_logtest_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_manager_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_rbac_white_mitre_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_overview_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_rootcheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings
	 
test_rbac_white_sca_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_task_endpoints.tavern.yaml 
	 1 passed, 3 warnings
	 
test_rootcheck_endpoints.tavern.yaml 
	 4 passed, 6 warnings

test_sca_endpoints.tavern.yaml 
	 3 passed, 5 warnings

test_default_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_all_endpoints.tavern.yaml 
	 148 passed, 3 xfailed, 153 warnings

test_task_endpoints.tavern.yaml 
	 2 passed, 4 warnings
```

Regards,
Manuel.

--- a/api/test/integration/env/configurations/vulnerability/manager/configuration_files/ossec_module.conf
+++ b/api/test/integration/env/configurations/vulnerability/manager/configuration_files/ossec_module.conf
@@ -1,7 +1,7 @@
 <root>
     <vulnerability-detector>
         <enabled>yes</enabled>
-        <interval>1h</interval>
+        <interval>1m</interval>
         <ignore_time>6h</ignore_time>
         <run_on_start>yes</run_on_start>
 

--- a/api/test/integration/tavern_utils.py
+++ b/api/test/integration/tavern_utils.py
@@ -56,7 +56,8 @@ def test_select_key_affected_items(response, select_key):
         set1 = main_keys.symmetric_difference(set(item.keys()))
         # Check if there are keys in response that were not specified in 'select_keys', apart from those which can be
         # mandatory (id, agent_id, etc).
-        assert (set1 == set() or set1 == set1.intersection({'id', 'agent_id', 'file'} | main_keys)), \
+        assert (set1 == set() or set1 == set1.intersection({'id', 'agent_id', 'file', 'task_id', 'agent_id', 'status',
+                                                            'command', 'create_time'} | main_keys)), \
             f'Select keys are {main_keys}, but the response contains these keys: {set1}'
 
         for nested_key in nested_keys.items():

